sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp

```diff
@@ -26,8 +26,16 @@ KeyClient::KeyClient(
   policies.emplace_back(std::make_unique<TelemetryPolicy>("KeyVault", apiVersion));
   policies.emplace_back(std::make_unique<RequestIdPolicy>());
   policies.emplace_back(std::make_unique<RetryPolicy>(options.RetryOptions));
-  policies.emplace_back(std::make_unique<BearerTokenAuthenticationPolicy>(
-      credential, "https://vault.azure.net/.default"));
+
+  {
+    Azure::Core::GetTokenOptions const tokenOptions
+        = {{"https://vault.azure.net/.default"},
+           std::make_unique<Azure::Core::Http::TransportPolicy>(options.TransportPolicyOptions)};
```

Comment:
Does this need to be mandatory?

Reply:
I think it is better to have it mandatory. Tecnically, you don't have to initialize it, but then it will be null, and nothing good would happen. We can always add a logic that if `GetTokenOptions::TransportPolicy` is null, then use `Details::GetTransportAdapter()`, but I don't think it is good if client libraries would do that. It is up to client library to put a non-null value there, the same way that they do it for their own `TransportPolicy`. So, this does not prevent client libs from using the default - either here or for their own `TransportPolicyOptions`, we're just saying don't give us null, initialize the at least the same way you initialize your own `TransportPolicyOptions`, do it consciously, or we may end up with the clients that have their own HTTP transport customizable, but the transport for authentication would always be using the default value, and that would be a miss.